    @Test
    @RunTestInLooperThread
    public void executeTransactionAsync_onError() throws Throwable {
        final Realm realm = looperThread.realm;
        assertEquals(0, realm.allObjects(Owner.class).size());

        realm.executeTransactionAsync(new Realm.Transaction() {
            @Override
            public void execute(Realm realm) {
                throw new RuntimeException("Oh! What a Terrible Failure");
            }
        }, new Realm.Transaction.OnError() {
            @Override
            public void onError(Throwable error) {
                assertEquals(0, realm.allObjects(Owner.class).size());
                assertNull(realm.where(Owner.class).findFirst());
                looperThread.testComplete();
            }
        });
    }

